libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt

```diff
@@ -609,6 +609,20 @@ class MapboxNavigationTest {
         mapboxNavigation.onDestroy()
     }
 
+    @Test
+    fun `updateLegIndex should return true when route leg is updated`() {
+        every { tripSession.updateLegIndex(2) } returns true
+
+        assertTrue(mapboxNavigation.updateLegIndex(2))
+    }
+
+    @Test
+    fun `updateLegIndex should return false when route leg is not updated`() {
+        every { tripSession.updateLegIndex(3) } returns false
+
+        assertFalse(mapboxNavigation.updateLegIndex(3))
+    }
```

Comment:
```suggestion
    @Test
    fun `updateLegIndex should return false when set negative leg index`() {
        every { tripSession.updateLegIndex(-1) } returns false

        assertFalse(mapboxNavigation.updateLegIndex(-1))
    }
```

Reply:
I don't see how is that any different. There's no logic in `MapboxNavigation.updateLegIndex`, that depends on the sign of the `legIdex` argument.